### PR TITLE
Add generic GetProperty on nodes and relationships

### PR DESCRIPTION
**Pull request by fbiville**
These are generic helpers to extract properties from nodes and relationships in a type-safe way.

Relates to #373

### CODE REVIEW

**Review by AndyHeap-NeoTech — APPROVED** [review]
Useful additions!

File: neo4j/dbtype/graph.go [hunk 1]
```diff
@@ -44,6 +63,18 @@ type Relationship struct {
...
+	return r.Id
+}
+
+func (r Relationship) GetElementId() string {
+	return r.ElementId
+}
+
+func (r Relationship) GetProperties() map[string]any {
+	return r.Props
+}
```
Comment (bigmontz) [inline neo4j/dbtype/graph.go:76]: Are this changes related with add generic to GetProperty?

**Review by bigmontz — APPROVED** [review]
⚙️

**Review by AndyHeap-NeoTech — APPROVED** [review]
Nice refinement to the original PR.

File: neo4j/dbtype/graph.go (hunk 1 above)
Comment (fbiville) [reply neo4j/dbtype/graph.go:76]: `GetProperties` is called in `GetProperty[T]` as follows:
```
rawValue, found := entity.GetProperties()[key]
```
I added the rest since Id/ElementId is kinda what constitutes the minimum definition of a Neo4j entity but it's not strictly needed.

**Review by bigmontz — APPROVED** [review]
🧂
